### DISCUSSION TITLE
model reuse between different projects?

### DISCUSSION THREAD

**Discussion by brianbrownton** [sequelize/sequelize#17064] [Help & Questions]
If I have Project A where I have defined a User model with many attributes, what is the recommended way to use this same model in Project B?

In reality they are not entirely separate projects, but both part of a larger project that have areas of overlap. An example would be a member experience and admin experience that both would need access to the same "User" model.

**Comment by ephys** [comment]
You could export common code from a git submodule, or from an internal npm package

If you use the `sequelize.define` or `model.init` approaches, the package could export functions that accept the sequelize instance and define the models

If you use `sequelize-typescript` or sequelize 7's decorators, you can just export the decorated model from the package, and add them to your sequelize instances in each project

**Reply by brianbrownton** [reply]
This is a good suggestion, thank you. 

Any guidance on then managing migrations between multiple projects?

**Reply by ephys** [reply]
Do all projects use the same database or are they different databases?

**Reply by brianbrownton** [reply]
Same database.
